packages/compass-data-modeling/src/components/diagram-editor.tsx

```diff
@@ -124,7 +132,21 @@ const DiagramEditor: React.FunctionComponent<{
   onCancelClick,
   onApplyClick,
 }) => {
+  const isDarkMode = useDarkMode();
   const [applyInput, setApplyInput] = useState('{}');
+  const diagram = useDiagram();
+
+  // For tests, we are setting the diagram instance on the window object
+  // so that we can access it in the tests.
+  useEffect(() => {
+    if (
+      process.env.APP_ENV === 'webdriverio' ||
+      process.env.NODE_ENV === 'development'
+    ) {
+      (window as any).diagramInstance = diagram;
+    }
+  }, [diagram]);
```

Comment:
Instead of using global window and gating it to these very specific checks (that we really truly should try to avoid), let's do something similar to what we do in codeedior and attach the diagram to the relevant DOM node that we can find in e2e tests, something like:

```
<div
  data-testid="diagram-editor-container"
  ref=(ref => {
    if (ref) {
      ref.__diagram = diagram;
    }
  })
```